src/pint/scripts/event_optimize.py

```diff
@@ -759,16 +766,17 @@ def main(argv=None):
     import emcee
 
     # Following are for parallel processing tests...
-    if 0:
+    if numcores != 1:
 
-        def unwrapped_lnpost(theta, ftr=ftr):
+        def unwrapped_lnpost(theta):
             return ftr.lnposterior(theta)
 
         import pathos.multiprocessing as mp
 
-        pool = mp.ProcessPool(nodes=8)
+        pool = mp.ProcessPool(nodes=numcores)
```

Comment:
If possible it is better to use `ProcessPool` in a `with` statement to ensure that everything gets shut down correctly even if an exception occurs.

Reply:
Should be updated now.

Reply:
Uh, have you tried it? I don't think it will work - you need the actual chain generation to be within the `with` statement, as soon as you exit that all the cores shut down. Sorry if the actual computation happens somewhere different than I thought...

Reply:
I tried it with the chain generation inside and outside the `with` statement and both appeared to work. I have changed it so that the chain generation is within the `with` statement now